ipc/src/main/java/io/scalecube/ipc/ChannelContext.java

```diff
@@ -71,59 +69,56 @@ public Address getAddress() {
   }
 
   public Observable<Event> listen() {
-    return eventSubject.onBackpressureBuffer().asObservable();
+    return subject.onBackpressureBuffer().asObservable();
   }
 
-  public Observable<Void> listenClose() {
-    return closeSubject.onBackpressureBuffer().asObservable();
-  }
-
-  public Observable<Event> listenMessageReceived() {
+  public Observable<Event> listenReadSuccess() {
     return listen().filter(Event::isReadSuccess);
   }
 
-  public Observable<Event> listenReadError() {
-    return listen().filter(Event::isReadError);
-  }
-
   public Observable<Event> listenMessageWrite() {
     return listen().filter(Event::isMessageWrite);
   }
 
   public void postReadSuccess(ServiceMessage message) {
-    eventSubject.onNext(new Event.Builder(ReadSuccess, this).message(message).build());
+    subject.onNext(new Event.Builder(ReadSuccess, this).message(message).build());
   }
 
   public void postReadError(Throwable throwable) {
-    eventSubject.onNext(new Event.Builder(ReadError, this).error(throwable).build());
+    subject.onNext(new Event.Builder(ReadError, this).error(throwable).build());
   }
 
   public void postMessageWrite(ServiceMessage message) {
-    eventSubject.onNext(new Event.Builder(MessageWrite, this).message(message).build());
+    subject.onNext(new Event.Builder(MessageWrite, this).message(message).build());
```

Comment:
this function postMessageWrite looks exactlly like postReadSuccess
do we really need 2?